_meeting-notes/2019-12-03.md

```diff
@@ -0,0 +1,168 @@
+        -~~ AI: make this discoverable! (David)~~ (moved to Assignments, below)
+    - @raulk - ASK: to @stebalien: could you write a design blueprint for traffic shaping? Let's make the current behaviour the starting point. Further enhancements / changes would be patches on top of that doc.
+        - ~~YES! AI: Steven said he can do it. <- push it to the docs folder~~ (moved to Assignments, below)
+
+
+## :dart: Next
+
+~~### OKRs for the Quarter~~ #1 Goal: Unblock go-ipfs 0.5.0
+~~https://raw.githubusercontent.com/ipfs/testground/master/docs/img/Q4OKRs.png~~
+~~### Top 3 things to get done this week~~
+
+
+
+We are putting aside the OKRs for the time being and instead **focus on the #1 Goal: Unblock go-ipfs 0.5.0**. This goal has a very clear list of things that need to ship:
+
+- [ ] Instrument go-libp2p so that the DHT operations are observable and traced in a Distributed Context (owner: @raulk)
+- [ ] Add support to Traffic Shaping & Network Configuration to TestGround (owner: @Stebalien)
+- [ ] Expand the TestGround Infrastructure so that it can run multiple thousands of nodes (owner: @nonsense)
+- [ ] Creation of a TestGround Dashboard that can aid the developers of go-ipfs & go-libp2p to grok the results outputted by every node in a TestGround run (owner: @daviddias)
+- [ ] Keep on iterating the DHT Test Plan (@owner: @daviddias)
+- [ ] Design blueprint for multi-version tests (aka test ensembles) (owner: @raulk)
+
+Full list can be at https://github.com/ipfs/testground/issues/196
+
+
+### Assignments
+
+- ==@raulk==
+    - Review daemon PR.
+- ==@daviddias==
+    - I'll have to dedicate most of my week to:
+        - Life Admin
+        - ResNetLab Deliverables (so that I can be fully heads down during the Hack Week)
+    - For TestGround specifically
+        - Continue preparing the Hack Week
+        - Collect architecture decisions that need to be captured and written down
+        - Create an Epic that tracks the Dependency Graph Tracking and link it to the go-ipfs 0.5.0 epic
+        - Link to HOW_WE_WORK from CONTRIBUTING.md
+- ==@stebalien==
+    - go-ipfs Hack Week
+    - For TestGround
+        - Last few changes on sidecar (the jitter PR + blocking inbound connection)
+        - Follow up on Jim's suggestion for Network Isolation (NAT) https://github.com/ipfs/testground/issues/96#issuecomment-559888146
+        - Write a Design Blueprint for Traffic Shapping (take inspiration from https://github.com/ipfs/testground/pull/203)
+- ==@jimpick==
+    - Test Plan 4
+    - Test SideCar and demonstrate how to use it (docs or recorded demo)
+- ==@nonsense==
+    - Merge the Infra PR from Jim
+    - Get the daemon PR merged
+- ==@hacdias==
+    - Dependency Graph Tracking https://github.com/ipfs/testground/issues/147
+    - Chew Large Datasets Plan
```

Comment:
@hacdias can you unpack what you plan to do on this Test Plan for this week? There are multiple test cases that you could target, it is wiser to focus on a few at a time :)

Reply:
I was thinking about upgrading the tests we already have by supporting multiple file sizes _and_ nested directories first, and then move to the others since I'd already have a structure for them. Sounds good? I can update the notes.

Reply:
That sounds great! it would be great if you could get the one with sharding experiment enabled as well

Reply:
Ah yes, I did enable the sharding thingy already... Forgot to mention that haha

Reply:
I just need to add the file!

Reply:
Updated!

Reply:
Note that sharding is for directories! :)